docs/source/plugins.rst

```diff
@@ -0,0 +1,137 @@
+=======
+Plugins
+=======
+
+Rez is designed around the concept of plugins. Plugins can be used to extend rez's functionalities without modifying any of rez's source code.
+
+Plugins are currently bundled in the main rez repo, but will be split out
+to their own repos in the future.
+
+The built-in plugins are located at :gh-rez:`src/rezplugins`.
```

Comment:
```suggestion
Rez comes with built-in plugins that are located at :gh-rez:`src/rezplugins`. New plugins are encouraged to be developed out-of-tree (outside rez).
```
It is true that we want to maybe split them out (at least some of them), but we didn't make any decision yet on this.